environment/src/main/java/com/sequenceiq/environment/parameters/validation/validators/parameter/AzureParameterValidator.java

```diff
@@ -144,11 +144,29 @@ private ValidationResult validateEntitlement(ValidationResultBuilder validationR
                         build();
             }
         }
+        String encryptionKeyResourceGroupName = azureResourceEncryptionParametersDto.getEncryptionKeyResourceGroupName();
+        if (Objects.nonNull(encryptionKeyResourceGroupName)) {
+            if (!entitlementService.isAzureDiskSSEWithCMKEnabled(accountId)) {
+                LOGGER.info("Invalid request, CDP_CB_AZURE_DISK_SSE_WITH_CMK entitlement turned off for account {}", accountId);
+                return validationResultBuilder.error(
+                        "You specified encryptionKeyResourceGroupName to provide the resource group name which contains the encryption key" +
+                                "for Server Side Encryption of Azure Managed disks, but that feature is currently disabled. " +
+                                "Get 'CDP_CB_AZURE_DISK_SSE_WITH_CMK' enabled for your account to use SSE with CMK.").
+                        build();
+            }
+            if (!Objects.nonNull(encryptionKeyUrl)) {
+                LOGGER.info("Invalid request, encryptionKeyResourceGroupName cannot be specified without encryptionKeyUrl");
+                return validationResultBuilder.error(
+                        "You specified encryptionKeyResourceGroupName to provide the resource group name which contains the encryption key for Server Side Encryption" +
+                                "of Azure Managed disks. Please specify encryptionKeyUrl to use Server Side Encryption for Azure Managed disks with CMK.").
+                        build();
+            }
```

Comment:
A debug log would be nice to have that the parameters look valid.

Reply:
Added a debug logline `LOGGER.debug("Validation of encryption parameters is successful.")`